### ISSUE TITLE
tvguide.co.uk grabs now fail

### ISSUE THREAD

**Issue by misar1** [K-vanc/Tempest-EPG-Generator#44]
The appearance of the browser EPG recently changed and it seems this has broken the existing site configuration. All channels now fail to return any programme information. Could you investigate please.

**Comment by K-vanc** [comment]
What this means??

At least you can attach some screenshot or etc. I cannot guess your every problem..

"The appearance of the browser EPG recently changed and it seems this has broken the existing site configuration. "

**Comment by K-vanc** [comment]
Do you mean website changed?

**Comment by K-vanc** [comment]
Ok So website changed. I just made a new release(rv1.5.3) of Tempest today. I thought something has broken on your browser after update :)

You can simply say "Website changed" or better only "not working" as everyone does :)  Anyway, urls are still responding but probably data structure changed. I will try to have a look tomorrow.

**Comment by misar1** [comment]
[                   ]-------------------------------------
[                   ]        Tempest EPG Generator
[                   ]
[                   ]   The Most Advanced Programmable
[                   ]       GUI XMLTV EPG Generator
[                   ]     by Kivanc Altug aka "Kvanc"
[                   ]-------------------------------------
[                   ]
[06-09-2023 21:23:40] Tempest Version : rv1.5.2
[06-09-2023 21:23:40] System Info : Windows NT SWIFT_EDGE 10.0 build 22621 (Windows 11) AMD64
[06-09-2023 21:23:40] PHP Version : 8.2.3
[06-09-2023 21:23:40] Server Info : Command Line Interface (Cli)
[06-09-2023 21:23:40] Initializing...
[06-09-2023 21:23:40] Checking required modules...
[06-09-2023 21:23:40] Time converter enabled...
[06-09-2023 21:23:40] tvguideUK_tempest-8.xml generated...
[06-09-2023 21:23:40] Grabbing started for 8 days
[                   ]
[06-09-2023 21:23:40] (1/58) Site: [ENC][EX]tvguide.co.uk_0
[06-09-2023 21:23:40] Channel id: 506
[06-09-2023 21:23:40] Xmltv id: 4Music
[      Warning      ]
[06-09-2023 21:23:41] No show grabbed in 0.63 seconds
[                   ]
[06-09-2023 21:23:41] (2/58) Site: [ENC][EX]tvguide.co.uk_0
[06-09-2023 21:23:41] Channel id: 752
[06-09-2023 21:23:41] Xmltv id: 4seven
[      Warning      ]
[06-09-2023 21:23:42] No show grabbed in 0.53 seconds

----------------------------------------------------------------
Etc for all 58 channels.

The comment about the website changing was additional information which may be relevant.

**Comment by K-vanc** [comment]
I uploaded revised siteconfig into UK folder. It seems they have changed everything so you have to replace all existing channels with new ones. Also website changed into detail page mode now so it will grab slower than before.

**Comment by misar1** [comment]
Thank you. Now working again but as you said much slower.

[                   ]
[                   ]-------------------------------------
[                   ]        Tempest EPG Generator
[                   ]
[                   ]   The Most Advanced Programmable
[                   ]       GUI XMLTV EPG Generator
[                   ]     by Kivanc Altug aka "Kvanc"
[                   ]-------------------------------------
[                   ]
[07-09-2023 10:25:43] Tempest Version : rv1.5.2
[07-09-2023 10:25:43] System Info : Windows NT SWIFT_EDGE 10.0 build 22621 (Windows 11) AMD64
[07-09-2023 10:25:43] PHP Version : 8.2.3
[07-09-2023 10:25:43] Server Info : Command Line Interface (Cli)
[07-09-2023 10:25:43] Initializing...
[07-09-2023 10:25:43] Checking required modules...
[07-09-2023 10:25:43] Time converter enabled...
[07-09-2023 10:25:43] tvguideUK_tempest-8.xml generated...
[07-09-2023 10:25:43] Grabbing started for 8 days
[                   ]
[07-09-2023 10:25:43] (1/58) Site: [ENC][EX]tvguide.co.uk_0
[07-09-2023 10:25:43] Channel id: a4539d2a-1cee-52f5-a00e-d68835ce3e9f/4music
[07-09-2023 10:25:43] Xmltv id: 4Music
[07-09-2023 10:28:20] 108 shows grabbed in 157.11 seconds
[                   ]
[07-09-2023 10:28:20] (2/58) Site: [ENC][EX]tvguide.co.uk_0
[07-09-2023 10:28:20] Channel id: 3d8bb828-1eb5-5de5-bb40-efba8bae9835/4seven
[07-09-2023 10:28:20] Xmltv id: 4seven
[07-09-2023 10:32:08] 201 shows grabbed in 227.72 seconds
[                   ]

**Comment by misar1** [comment]
Thanks again. I forgot about the update and also about  index-only which is about 50x faster.
[                   ]
[                   ]-------------------------------------
[                   ]        Tempest EPG Generator
[                   ]
[                   ]   The Most Advanced Programmable
[                   ]       GUI XMLTV EPG Generator
[                   ]     by Kivanc Altug aka "Kvanc"
[                   ]-------------------------------------
[                   ]
[07-09-2023 12:25:32] Tempest Version : rv1.5.3
[07-09-2023 12:25:32] System Info : Windows NT SWIFT1 10.0 build 19045 (Windows 10) AMD64
[07-09-2023 12:25:32] PHP Version : 8.2.3
[07-09-2023 12:25:32] Server Info : Command Line Interface (Cli)
[07-09-2023 12:25:32] Initializing...
[07-09-2023 12:25:32] Checking required modules...
[07-09-2023 12:25:32] Time converter enabled...
[07-09-2023 12:25:32] tvguideUK_tempest-8.xml generated...
[07-09-2023 12:25:32] Grabbing started for 8 days
[                   ]
[07-09-2023 12:25:32] (1/58) Site: [ENC][EX]tvguide.co.uk_0
[07-09-2023 12:25:32] Channel id: a4539d2a-1cee-52f5-a00e-d68835ce3e9f/4music
[07-09-2023 12:25:32] Xmltv id: 4Music
[07-09-2023 12:25:37] 108 shows grabbed in 4.80 seconds
[                   ]
[07-09-2023 12:25:37] (2/58) Site: [ENC][EX]tvguide.co.uk_0
[07-09-2023 12:25:37] Channel id: 3d8bb828-1eb5-5de5-bb40-efba8bae9835/4seven
[07-09-2023 12:25:37] Xmltv id: 4seven
[07-09-2023 12:25:42] 201 shows grabbed in 5.66 seconds
[                   ]

**Comment by K-vanc** [comment]
yes, it is much more faster but in return you have to sacrifice some xmltv elements. So it is matter of choice. Anyway, enjoy

**Comment by misar1** [comment]
tvguide.co.uk was working correctly after you changed the configuration but there is a problem again. Many channels return no programmes and most others fail to find them for every day when set to 8 days. The example below is with index-only on but turning it off gives the same problem. The website guide is working correctly in a browser.

<?xml version="1.0" encoding="UTF-8"?>

<configuration>
  <filename>tvguideUK_tempest-8</filename>
  <user-agent/>
  <logger>off</logger>
  <retry>1</retry>
  <time-out>10</time-out>
  <timespan>7</timespan>
  <proxy/>
  <index-only>on</index-only>
  <fusion>off</fusion>
  <fusion-level/>
  <time-converter>GMT</time-converter>
  <delay channel="" index="" detail=""/>
  <channel-lister>off</channel-lister>
  <inverter>off</inverter>
  <inverter-filename/>
  <inverter-path/>

[                   ]
[                   ]-------------------------------------
[                   ]        Tempest EPG Generator
[                   ]
[                   ]   The Most Advanced Programmable
[                   ]       GUI XMLTV EPG Generator
[                   ]     by Kivanc Altug aka "Kvanc"
[                   ]-------------------------------------
[                   ]
[20-09-2023 05:41:04] Tempest Version : rv1.5.2
[20-09-2023 05:41:04] System Info : Windows NT SWIFT_EDGE 10.0 build 22621 (Windows 11) AMD64
[20-09-2023 05:41:04] PHP Version : 8.2.3
[20-09-2023 05:41:04] Server Info : Command Line Interface (Cli)
[20-09-2023 05:41:04] Initializing...
[20-09-2023 05:41:04] Checking required modules...
[20-09-2023 05:41:04] Time converter enabled...
[20-09-2023 05:41:04] tvguideUK_tempest-8.xml generated...
[20-09-2023 05:41:04] Grabbing started for 8 days
[                   ]
[20-09-2023 05:41:04] (1/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:04] Channel id: a4539d2a-1cee-52f5-a00e-d68835ce3e9f/4music
[20-09-2023 05:41:04] Xmltv id: 4Music
[      Warning      ]
[20-09-2023 05:41:05] No show grabbed in 1.14 seconds
[                   ]
[20-09-2023 05:41:05] (2/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:05] Channel id: 3d8bb828-1eb5-5de5-bb40-efba8bae9835/4seven
[20-09-2023 05:41:05] Xmltv id: 4seven
[20-09-2023 05:41:10] 85 shows grabbed in 4.68 seconds
[                   ]
[20-09-2023 05:41:10] (3/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:10] Channel id: 022ec304-25b9-5b07-aaf1-c90a0393abce/5action
[20-09-2023 05:41:10] Xmltv id: 5Action
[20-09-2023 05:41:13] 49 shows grabbed in 2.76 seconds
[                   ]
[20-09-2023 05:41:13] (4/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:13] Channel id: f3a9b277-2bf1-5755-a8ea-7912e8063b01/5select
[20-09-2023 05:41:13] Xmltv id: 5SELECT
[20-09-2023 05:41:16] 70 shows grabbed in 3.27 seconds
[                   ]
[20-09-2023 05:41:16] (5/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:16] Channel id: 9512892c-d864-513c-b630-abb5df6b49f4/5star
[20-09-2023 05:41:16] Xmltv id: 5STAR
[      Warning      ]
[20-09-2023 05:41:17] No show grabbed in 0.92 seconds
[                   ]
[20-09-2023 05:41:17] (6/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:17] Channel id: 4b75d2ae-672b-5d6f-b1d8-a0b2761fde3a/5usa
[20-09-2023 05:41:17] Xmltv id: 5USA
[      Warning      ]
[20-09-2023 05:41:18] No show grabbed in 0.92 seconds
[                   ]
[20-09-2023 05:41:18] (7/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:18] Channel id: ced32d78-bb93-5d4b-ba79-b8cfdc79fd58/al-jazeera-english
[20-09-2023 05:41:18] Xmltv id: Al Jazeera English
[      Warning      ]
[20-09-2023 05:41:20] No show grabbed in 1.53 seconds
[                   ]
[20-09-2023 05:41:20] (8/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:20] Channel id: f2fb2e91-2578-5975-a7cc-dd048186ea52/bbc-four
[20-09-2023 05:41:20] Xmltv id: BBC Four
[20-09-2023 05:41:22] 12 shows grabbed in 2.36 seconds
[                   ]
[20-09-2023 05:41:22] (9/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:22] Channel id: 0e5222b1-5665-5d26-855e-6a81768dcfa1/bbc-news
[20-09-2023 05:41:22] Xmltv id: BBC News
[      Warning      ]
[20-09-2023 05:41:23] No show grabbed in 1.43 seconds
[                   ]
[20-09-2023 05:41:23] (10/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:23] Channel id: 78bd54a7-6883-575e-a84f-f328dc424979/bbc-one
[20-09-2023 05:41:23] Xmltv id: BBC One
[      Warning      ]
[20-09-2023 05:41:25] No show grabbed in 1.43 seconds
[                   ]
[20-09-2023 05:41:25] (11/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:25] Channel id: 7735d86e-bf3c-55ba-b83f-2e110e9e96f6/bbc-parliament-hd
[20-09-2023 05:41:25] Xmltv id: BBC Parliament HD
[      Warning      ]
[20-09-2023 05:41:26] No show grabbed in 0.92 seconds
[                   ]
[20-09-2023 05:41:26] (12/58) Site: [ENC][EX]tvguide.co.uk_0
[20-09-2023 05:41:26] Channel id: 90230fa8-b213-5bda-b8ac-0ea968795c73/bbc-red-button-1
[20-09-2023 05:41:26] Xmltv id: BBC Red Button 1
[20-09-2023 05:41:27] 17 shows grabbed in 1.74 seconds
[                   ]
Etc

**Comment by K-vanc** [comment]
there is no problem with siteconfig, if you check website for the CHANNEL ARE NOT WORKING, you will see there is no epg, more even channels are not exist and if you force the open them, you will see something as below(example for 4Music);

![image](https://github.com/K-vanc/Tempest-EPG-Generator/assets/97025515/ed202634-f0a5-4dbb-bd55-4bb8342fd45b)

Sorry but Tempest is a grabber, it will grab epg if there is data in the website, unfortunately, it cannot prepare random epg when there is no data

**Comment by K-vanc** [comment]
I dont think that you understand what I mean. These channels dont have data currently at website. The channels have data in the website, you are already able to grab with Tempest or with some other tool. Lets check your file for 4Music again which is currectly no data at the moment..

![image](https://github.com/K-vanc/Tempest-EPG-Generator/assets/97025515/48f43610-fa53-4197-a6fe-ee384a0555ae)

Did you see number of counts?? Only 1, so no epg for this channel also on your other tool..

**Comment by K-vanc** [comment]
ok, i see. I checked your tool and it is using old version of tvguide.co.uk api. Such api is not available in website. So we are not trying to get  data from same place exactly. I dont know how that is still working because when I try, it redirects me to now url and give no data for that empty channels

**Comment by misar1** [comment]
I do not know how either grabber works but thanks for the explanation. 

The problem may be caused by the complexity of the new website. It has a very large number of options for different UK guides  ("broadcasters") and different UK locations for each guide. 4Music is not displayed with every guide because it is not available with that broadcaster. For example it is not listed for Freeview but appears as channel 354 for Sky. Similar problems arise with other channels. The most comprehensive guide is Sky so if you can set your grabber to use that with London it should find almost every available UK channel. The alternative would be to add options in the .config to set  both guide and location.

**Comment by K-vanc** [comment]
When it changes with the location, I am adding option such as userkey. For this website it is not working as you stated. No matter 1 channel,  available in several different providers, channel id is all same. thats why I created single list..

And, yes, 4music is exist under sky tab but when you click to see channel, this happens again.. 

![image](https://github.com/K-vanc/Tempest-EPG-Generator/assets/97025515/89b8c4fc-ce5d-4abc-af90-5172b9b98de7)

and when you click

![image](https://github.com/K-vanc/Tempest-EPG-Generator/assets/97025515/4e94a24b-a103-40db-95bc-bb75f2746722)

so please be patient till website fix their error and you will be able to grab again

**Comment by K-vanc** [comment]
ok, so if you believe you understand more than me, please make your own siteconfig according to sky and use

**Comment by K-vanc** [comment]
I prepared SKY only version for you and this is the same with your other tool. I will upload this into my repo since it working in a way that I dont like. Since there is no way of filtering single channel on grid view, it will fetch  massive amount of data containing all channels and will not work on php's default memory settings. You need to edit your php.ini file for the memory size and restart your php server to take effect of new value.

memory_limit = 128M

above is the line for deault value which is 128MB, in your php.ini. Required memory size will increase with the quantity of days you requested. No worry to increase it, other tools also are draining the same memory, at least php is asking you before doing..

[[ENC][EX]tvguide.co.uk(SKY).zip](https://github.com/K-vanc/Tempest-EPG-Generator/files/12673602/ENC.EX.tvguide.co.uk.SKY.zip)

**Comment by K-vanc** [comment]
You are welcome. I have a very bad habit such as being right most of the time but unfortunately cannot convince people to it :) 

I am using/working on Tempest since 2019 and was team member of another grabber for another several years before that so i know enough to just step back and wait for a few days when you see a server error message on the browser before start digging in more.
